pkg/providers/aws.go

```diff
@@ -155,3 +198,108 @@ var _ = Describe("MetadataServiceOptions", framework.LabelCloudProviderSpecific,
 		assertIMDSavailability(machineSet, "HTTP_CODE:200")
 	})
 })
+
+var _ = Describe("CapacityReservationID", framework.LabelCloudProviderSpecific, framework.LabelProviderAWS, func() {
+	var client runtimeclient.Client
+	var gatherer *gatherer.StateGatherer
+	var ctx context.Context
+
+	toDelete := make([]*machinev1.MachineSet, 0, 3)
+
+	BeforeEach(func() {
+		var err error
+		client, err = framework.LoadClient()
+		Expect(err).ToNot(HaveOccurred(), "Failed to load client")
+
+		gatherer, err = framework.NewGatherer()
+		Expect(err).ToNot(HaveOccurred(), "Failed to load gatherer")
+
+		ctx = framework.GetContext()
+
+		platform, err := framework.GetPlatform(ctx, client)
+		Expect(err).ToNot(HaveOccurred(), "Failed to get platform")
+		if platform != configv1.AWSPlatformType {
+			Skip(fmt.Sprintf("skipping AWS specific tests on %s", platform))
+		}
+		// Make sure to clean up the resources we created
+		DeferCleanup(func() {
+			Expect(framework.DeleteMachineSets(client, toDelete...)).To(Succeed())
+			toDelete = make([]*machinev1.MachineSet, 0, 3)
+
+			framework.WaitForMachineSetsDeleted(ctx, client, toDelete...)
+		})
+	})
+
+	AfterEach(func() {
+		specReport := CurrentSpecReport()
+		if specReport.Failed() {
+			Expect(gatherer.WithSpecReport(specReport).GatherAll()).To(Succeed())
+		}
+	})
+
+	createMachineSet := func(capacityReservationId string) (*machinev1.MachineSet, error) {
```

Comment:
nit: I might make this more descriptive, e.g `createMachineSetWihCapacityReservationId`